src/ncp/spinel.h

```diff
@@ -1589,7 +1601,24 @@ typedef enum {
     SPINEL_PROP_STREAM_EXT__END   = 0x1800,
 
     SPINEL_PROP_MESHCOP__BEGIN = 0x80,
-    SPINEL_PROP_MESHCOP__END   = 0x90,
+
+    SPINEL_PROP_MESHCOP_JOINER_STATE = SPINEL_PROP_MESHCOP__BEGIN + 0, ///<[C]
+
+    /// Joiner commission command and the parameters
+    /** Format `bUU`
+     *
+     * This property starts or stops Joiner's commissioning process
+     *
+     * Data per item is:
+     *
+     *  `b` : Start or stop commissioning process
+     *  `U` : Joiner's PSKd if start commissioning
+     *  `U` : Provisioning url if start commissioning
+     *
```

Comment:
How about adding that this property is `write only` and maybe a comment here about the how the result of the joiner commissioning process is reported:

```
Writing to this property starts/stops the Joiner commissioning process. The immediate `VALUE_IS` response 
indicates success/failure of the starting/stopping the Joiner commissioning process.
After a successful start operation, the join process outcome is reported  through an asynchronous 
`VALUE_IS(LAST_STATUS)`  update with one of the following error status values:

- SPINEL_STATUS_JOIN_SUCCESS     the join process succeeded.
- SPINEL_STATUS_JOIN_SECURITY    the join process failed due to security credentials.
- SPINEL_STATUS_JOIN_NO_PEERS    no joinable network was discovered.
- SPINEL_STATUS_JOIN_RSP_TIMEOUT if a response timed out.
- SPINEL_STATUS_JOIN_FAILURE     join failure.
```